src/en/mangasaki/src/eu/kanade/tachiyomi/extension/en/mangasaki/MangaSaki.kt

```diff
@@ -0,0 +1,265 @@
+        }
+    }
+
+    override fun searchMangaSelector(): String {
+        return if (!searchMode) {
+            "div.view-content div.views-row"
+        } else {
+            "ol.search-results li.search-result"
+        }
+    }
+
+    override fun searchMangaFromElement(element: Element): SManga {
+        val Manga = SManga.create()
+        if (!searchMode) {
+            Manga.title = element.select("div.views-field-title a").text()
+            Manga.setUrlWithoutDomain(element.select("div.views-field-title a").attr("href"))
+            Manga.thumbnail_url = element.select("div.views-field-field-image2 img").attr("src")
+        } else {
+            // The site doesn't show thumbnails when using search
+            val titleElement = element.select("h3.title a")
+            Manga.title = titleElement.text()
+            Manga.setUrlWithoutDomain(titleElement.attr("href"))
+        }
+
+        return Manga
+    }
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    // manga details
+    override fun mangaDetailsParse(document: Document): SManga {
+        val Manga = SManga.create()
+        Manga.author = document.select("div.field-name-field-author div.field-item").first()?.text()
+        Manga.genre = document.select("div.field-name-field-genres ul li a").joinToString { it.text() }
+        Manga.description = document.select("div.field-name-body div.field-item p").text()
+        Manga.thumbnail_url = document.select("div.field-name-field-image2 div.field-item img").attr("src")
+
+        document.select("div.field-name-field-status div.field-item").text().also { statusText ->
+            when {
+                statusText.contains("Ongoing", true) -> Manga.status = SManga.ONGOING
+                statusText.contains("Complete", true) -> Manga.status = SManga.COMPLETED
+                else -> Manga.status = SManga.UNKNOWN
+            }
+        }
+
+        return Manga
+    }
+
+    // chapters
+    override fun chapterListRequest(manga: SManga) = chapterListRequest(manga.url, 1)
+
+    private fun chapterListRequest(url: String, page: Int): Request {
+        return GET("$baseUrl$url?page=${page - 1}", headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        var document = response.asJsoup()
+        val chapters = document.select(chapterListSelector()).map(::chapterFromElement).toMutableList()
+        var nextPage = 2
+
+        while (document.select(paginationNextPageSelector).isNotEmpty()) {
+            val dirtyPage = document.select("div#block-search-form form#search-block-form").attr("action")
+            val cleaningIndex = dirtyPage.lastIndexOf("?")
+            val cleanPage = dirtyPage.substring(0, cleaningIndex)
+            document = client.newCall(chapterListRequest(cleanPage, nextPage)).execute().asJsoup()
+            chapters += document.select(chapterListSelector()).map(::chapterFromElement)
```

Comment:
```suggestion
            chapters.addAll(document.select(chapterListSelector()).map(::chapterFromElement))
```

The `+` operator [allocates a new list](https://github.com/JetBrains/kotlin/blob/0938b46726b9c6938df309098316ce741815bb55/libraries/stdlib/common/src/generated/_Collections.kt#L3255-L3269).